Sources/SourceControl/CheckoutManager.swift

```diff
@@ -11,6 +11,15 @@
 import Basic
 import Utility
 
+/// Delegate to notify clients about actions being performed by CheckoutManager.
+public protocol CheckoutManagerDelegate: class {
```

Comment:
I think this delegate should be focused on the things only the class itself has internal ownership over, not reporting information that the client already knows.

Fetching makes sense to report, because it is a part of the core logic this class has responsibility over, but cloning is only something this class does in direct response to a client API call, so I don't think it belongs in this particular delegate.

Does that division make sense?